cmd/launcher/launcher.go

```diff
@@ -193,6 +193,9 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 	// we expect we're live. Record the version for osquery to
 	// pickup
 	internal.RecordLauncherVersion(rootDirectory)
+	if err = internal.RecordMetadata(rootDirectory, ctx, k); err != nil {
```

Comment:
Are we confident we'll have all this data on a fresh install since it has to come down via control server? Maybe that's okay and it will get written on the next launcher start up? or perhaps we would to do some retry logic in a goroutine.

Reply:
discussed offline^ my testing may have been flawed here- adding some async retry, thank you!

Reply:
Hrm... if we're really concerned about that, I wonder if we should move the write into the server data consumer.

Reply:
yea, this is better

Reply:
actually, a subscriber makes more sense